diff --git a/field_group.module b/field_group.module
index fc6aaba..94abf31 100755
--- a/field_group.module
+++ b/field_group.module
@@ -190,14 +190,15 @@ function field_group_theme_registry_alter(&$theme_registry) {
   }
 
   foreach ($entities as $entity) {
-    $theme_registry[$entity]['preprocess functions'][] = 'field_group_build_entity_groups';
-    // DS support, make sure it comes after field_group.
-    if ($key = array_search('ds_entity_variables', $theme_registry[$entity]['preprocess functions'])) {
-      unset($theme_registry[$entity]['preprocess functions'][$key]);
-      $theme_registry[$entity]['preprocess functions'][] = 'ds_entity_variables';
+    if(isset($theme_registry[$entity])) {	
+      $theme_registry[$entity]['preprocess functions'][] = 'field_group_build_entity_groups';
+      // DS support, make sure it comes after field_group.
+      if ($key = array_search('ds_entity_variables', $theme_registry[$entity]['preprocess functions'])) {
+        unset($theme_registry[$entity]['preprocess functions'][$key]);
+        $theme_registry[$entity]['preprocess functions'][] = 'ds_entity_variables';
+      }
     }
   }
-
 }
 
 /**
